tests/firedrake/regression/test_bddc.py

```diff
@@ -120,11 +134,44 @@ def mesh(request):
 
 
 @pytest.mark.parallel
-@pytest.mark.parametrize("family", "Q")
-@pytest.mark.parametrize("degree", (4,))
-@pytest.mark.parametrize("condense", (False, True))
-def test_bddc_fdm(mesh, family, degree, condense):
+@pytest.mark.parametrize("degree", range(1, 3))
+@pytest.mark.parametrize("variant", ("spectral", "fdm"))
+def test_vertex_dofs(mesh, variant, degree):
+    """Check that we extract the right number of vertex dofs from a high order Lagrange space."""
+    from firedrake.preconditioners.bddc import get_restricted_dofs
+    P1 = FunctionSpace(mesh, "Lagrange", 1, variant=variant)
+    V0 = FunctionSpace(mesh, "Lagrange", degree, variant=variant)
+    v = get_restricted_dofs(V0, "vertex")
+    assert v.getSizes() == P1.dof_dset.layout_vec.getSizes()
+
+
+@pytest.mark.parallel
+@pytest.mark.parametrize("family,degree", [("Q", 4)])
+def test_bddc_fdm(mesh, family, degree):
+    variant = "fdm"
+    bcs = True
+    tdim = mesh.topological_dimension
+    expected = 6 if tdim == 2 else 11
+    assert solve_riesz_map(mesh, family, degree, variant, bcs) <= expected
+
+
+@pytest.mark.parallel
+@pytest.mark.parametrize("family,degree", [("Q", 4)])
+@pytest.mark.parametrize("vector", (False, True), ids=("scalar", "vector"))
+def test_bddc_aij_quad(mesh, family, degree, vector):
+    variant = None
     bcs = True
     tdim = mesh.topological_dimension
     expected = 7 if tdim == 2 else 11
-    assert solve_riesz_map(mesh, family, degree, bcs, condense) <= expected
+    assert solve_riesz_map(mesh, family, degree, variant, bcs, vector=vector) <= expected
+
+
+@pytest.mark.parallel
+@pytest.mark.parametrize("family,degree", [("CG", 3), ("N1curl", 3), ("N1div", 3)])
+def test_bddc_aij_simplex(family, degree):
+    nx = 4
+    mesh = UnitCubeMesh(nx, nx, nx)
+    variant = None
+    bcs = True
+    expected = {"CG": 13, "N1curl": 14, "N1div": 12}[family]
```

Comment:
Where are these numbers from?

Reply:
Assuming a fixed mesh partition, of course